### ISSUE TITLE
X-Auth-CouchDB-* headers from original request are not overwritten

### ISSUE THREAD

**Issue by dstudzinski** [tyler-johnson/couchdb-auth-proxy#7]
Hi,

I have problem with headers. Original headers from original request (made from e.g. client, Postman app etc.) are not overwritten by couchdb-auth-proxy.

I think it's because in code headers are used as case-sensitive.
Based on info from this issue: [HTTP headers are being converted to lower case by http.js #1954](https://github.com/nodejs/node-v0.x-archive/issues/1954) headers are case-insensitive and node lowercases all headers (in req.headers ale headers are lowercase).

It means that request made with e.g. Postman with header like `X-Auth-CouchDB-UserName: someuser` comes to proxy as `x-auth-couchdb-username: someuser` and value is not overwritten by `req.headers[headerFields.username] = n` because default header is `X-Auth-CouchDB-UserName`.

**Header which comes to CouchDB has original value from Postman instead of value set in auth proxy.**

I think headerFields should be converted to lowercase before use (somewhere after `	headerFields = Object.assign({...`)

I've also tried a workaround - I configured headers by headerFields option. When I set them like this:
```
{
    "username": "x-auth-couchdb-username",
    "roles": "x-auth-couchdb-roles",
    "token": "x-auth-couchdb-token"
}
```
everything works as expected.

**Comment by tyler-johnson** [comment]
Okay yea this is definitely a bug. I haven't ever tried to pass existing proxy headers through so I never caught this.

I want to be absolute sure that those headers are removed before they are set, so I will fix this by removing the lowercase version of the headers.

Thanks!

**Comment by tyler-johnson** [comment]
Fixed in 1.1.3.  Thanks again!
